torch_xla/csrc/lowering_context.cpp

```diff
@@ -35,11 +38,21 @@ class HloMetadataSetter {
 
   static void PopulateXlaOpMetadata(LoweringContext* loctx, const Node* node) {
     xla::OpMetadata metadata;
-    metadata.set_op_type(node->op().ToString());
+    // NOTE: we apply some string manipulation as xprof backend utility
+    // for nesting/grouping traces depends on certain op name/type
+    // patterns for classification.
+    // https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/profiler/utils/tf_op_utils.cc#L55
+    std::string op_type =
+        absl::StrReplaceAll(node->op().ToString(), {{":", "_"}});
+    metadata.set_op_type(op_type);
     const ir::MetaData& nmeta = node->metadata();
+    std::string op_name_prefix;
     if (!nmeta.scope.empty()) {
-      metadata.set_op_name(nmeta.scope);
+      op_name_prefix =
+          absl::StrCat(absl::StrReplaceAll(nmeta.scope, {{":", "_"}}), "/");
     }
+    metadata.set_op_name(absl::StrCat(op_name_prefix, op_type));
```

Comment:
It seems like metadata will include the scope as a prefix, how would that be displayed on the GUI? It will be a nested graph or it is just that op name will be longer?

Reply:
Exactly, the scopes will appear as nested traces. The stack for xprof plugin:
* https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/profiler/utils/tf_op_utils.cc#L98-L104
* https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/profiler/utils/derived_timeline.cc#L78-L82